### PR TITLE
[Merged by Bors] - feat: comonoid objects in a braided category form a monoidal category

### PR DESCRIPTION
**Pull request by kim-em**
---

- [x] depends on: #10091
- [x] depends on: #10094 
- [x] depends on: #10095

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Monoidal/Comon_.lean [hunk 1]
```diff
@@ -227,10 +230,51 @@ def Mon_op_op_to_Comon : (Mon_ (Cᵒᵖ))ᵒᵖ ⥤ Comon_ C where
 /--
 Comonoid objects are contravariantly equivalent to monoid objects in the opposite category.
 -/
+@[simps]
 def Comon_equiv_Mon_op_op : Comon_ C ≌ (Mon_ (Cᵒᵖ))ᵒᵖ :=
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Monoidal/Comon_.lean:234]: Should not the capitalization of these definitions be slighly different, like `Comon_EquivMon_OpOp`?

Comment (semorrison) [reply Mathlib/CategoryTheory/Monoidal/Comon_.lean:234]: I have no idea how the naming rules are mean to apply when there are `_`s in the names. I agree that what I have must be wrong, but your suggestion seems terrible too. :-)

Comment (semorrison) [reply Mathlib/CategoryTheory/Monoidal/Comon_.lean]: Okay, changed to follow the pattern in your suggestion. I guess it is consistent!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**